javascript/extractor/src/com/semmle/js/extractor/YAMLExtractor.java

```diff
@@ -226,16 +226,19 @@ private String getTag(String explicitTag, NodeId kind, String value, boolean imp
    * <i>n</i> characters, where <i>n</i> is the number of spaces preceding the caret character on
    * the second line.
    *
-   * <p>This is only an approximation, since the context never extends across newlines, but it
-   * suffices for the purposes of <code>toString</code>.
+   * <p>This is only an approximation, since the context is limited to relatively short strings that
+   * never extend across newlines, but it suffices for the purposes of <code>toString</code>.
    */
   private String mkToString(Mark startMark, Mark endMark) {
     String snippet = startMark.get_snippet(0, Integer.MAX_VALUE);
     int nl = snippet.indexOf('\n');
     String context = snippet.substring(0, nl);
     String src = context.substring(snippet.substring(nl + 1).indexOf('^'));
-    if (endMark.getLine() == startMark.getLine())
-      src = src.substring(0, endMark.getColumn() - startMark.getColumn());
+    int desiredSingleLineSrcLength = endMark.getColumn() - startMark.getColumn();
+    boolean isShortString = src.length() >= desiredSingleLineSrcLength;
```

Comment:
I'm confused by the name `isShortString` when it's defined to be true is that `src` is *longer* than a certain limit. Was there a negation that was flipped during a refactoring here?

Reply:
Amended to `hasAccessToDesiredString`.